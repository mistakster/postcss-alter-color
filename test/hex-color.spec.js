const process = require('./utils/process');

it('should alter short hex colors', () => {
  const source = `
div {
  color: #000;
}
`;

  return process(source, {from: 'black', to: 'red'})
    .andMatchSnapshot();
});

it('should alter short hex to full hex colors', () => {
  const source = `
div {
  color: #000;
}
`;

  return process(source, {from: 'black', to: '#ff3fe0'})
    .andMatchSnapshot();
});

it('should alter short hex color in a complex value', () => {
  const source = `
div {
  color: #000;
  border: 1px solid #fff;
  outline: 1px solid #000
}
`;

  return process(source, {from: 'black', to: 'red'})
    .andMatchSnapshot();
});

it('should alter full hex colors', () => {
  const source = `
div {
  color: #000000;
}
`;

  return process(source, {from: 'black', to: 'red'})
    .andMatchSnapshot();
});

it('should alter full hex color in a complex value', () => {
  const source = `
div {
  color: #000000;
  border: 1px solid #ffffff;
  outline: 1px solid #000000;
}`;

  return process(source, {from: 'black', to: 'red'})
    .andMatchSnapshot();
});
