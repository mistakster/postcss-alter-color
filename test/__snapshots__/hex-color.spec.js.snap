// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should alter full hex color in a complex value 1`] = `
"
div {
  color: #ff0000;
  border: 1px solid #ffffff;
  outline: 1px solid #ff0000;
}"
`;

exports[`should alter full hex colors 1`] = `
"
div {
  color: #ff0000;
}
"
`;

exports[`should alter short hex color in a complex value 1`] = `
"
div {
  color: #f00;
  border: 1px solid #fff;
  outline: 1px solid #f00
}
"
`;

exports[`should alter short hex colors 1`] = `
"
div {
  color: #f00;
}
"
`;

exports[`should alter short hex to full hex colors 1`] = `
"
div {
  color: #ff3fe0;
}
"
`;
